### PR TITLE
Geomap: Documentation

### PR DESCRIPTION
**Pull request by euniceek**
**What this PR does / why we need it**:
This PR adds documentation for the Geomap plugin, addressing part of # 36585. 

**Special notes for your reviewer**:
Images are not included in grafana/website yet. Once those are added in, the images will load properly.

### CODE REVIEW

**Review by alolita — APPROVED** [review]

File: docs/sources/panels/visualizations/geomap.md [hunk 1]
```diff
@@ -8,4 +8,123 @@ weight = 600
...
+    "url": "https://tile.openstreetmap.org/{z}/{x}/{y}.png"
+  }
+}`
+```
+
+`enable_custom_baselayers` gives the option to enable/disable custom open source base maps that are already implemented. This will be defaulted to `true`.
+
+## Data layer
+
+Data layer in the Geomap plugin determine how you visualize geospatial data on top of the base map.
```
Comment (bryanuribe) [inline docs/sources/panels/visualizations/geomap.md]: determine --> determines

File: docs/sources/panels/visualizations/geomap.md [hunk 2]
```diff
@@ -8,4 +8,123 @@ weight = 600
...
+
+### Location
+The Geomap panel needs a source of geographical data. The data comes from a database query, and there are four mapping options for your data.
+
+- **Auto** will automatically search for location data. Use this data option when your query is based on one of the following names for data fields.
+  - geohash: “geohash”
+  - latitude: “latitude”, “lat”
+  - longitude: “longitude”, “lng”, “lon”
+  - lookup: “lookup”
+- **Coords** specifies that the query holds coordinate data. On selection, you will be prompted to select the data fields for latitude and longitude from your database query.
```
Comment (bryanuribe) [inline docs/sources/panels/visualizations/geomap.md]: - **Coords** specifies that the query holds coordinate data. On selection, you will be prompted to select the data fields for latitude and longitude from your database query.

Change to:

- **Coords** specifies that your query holds coordinate data. You will be prompted to select numeric data fields for latitude and longitude from your database query.

File: docs/sources/panels/visualizations/geomap.md [hunk 3]
```diff
@@ -8,4 +8,123 @@ weight = 600
...
+- **Geohash** specifies that the query holds geohash data. On selection, you will be prompted to select the data field for geohash from your database query.
```
Comment (bryanuribe) [inline docs/sources/panels/visualizations/geomap.md]: - **Geohash** specifies that the query holds geohash data. On selection, you will be prompted to select the data field for geohash from your database query.

Change to: 

- **Geohash** specifies that your query holds geohash data. You will be prompted to select a string data field for geohash from your database query.

File: docs/sources/panels/visualizations/geomap.md [hunk 4]
```diff
@@ -8,4 +8,123 @@ weight = 600
...
+- **Lookup** specifies that the query holds location name data that needs to be mapped to a value. On selection, you will be prompted to select the lookup field from your database query, and also the gazetteer. The gazeteer is the directory that will be used to map queried data to a geographical point.
```
Comment (bryanuribe) [inline docs/sources/panels/visualizations/geomap.md]: - **Lookup** specifies that the query holds location name data that needs to be mapped to a value. On selection, you will be prompted to select the lookup field from your database query, and also the gazetteer. The gazeteer is the directory that will be used to map queried data to a geographical point.

Change to: 

- **Lookup** specifies that your query holds location name data that needs to be mapped to a value. You will be prompted to select the lookup field from your database query and a gazetteer. The gazeteer is the directory that will be used to map your queried data to a geographical point.

**Pull request closed without merging**